From 45215af880c7898f495aabc3ddf69907e465ab34 Mon Sep 17 00:00:00 2001
From: Neal Gompa (ニール・ゴンパ) <ngompa13@gmail.com>
Date: Sat, 5 Dec 2015 23:19:26 -0500
Subject: [PATCH] Replace on_exit() with portable atexit()

The on_exit() function is non-portable and is not available
in all environments (notably Linux environments running on MUSL
and Solaris based environments). Thus, the function is swapped
for the portable atexit() call.
---
 src/createrepo_shared.c | 2 +-
 src/createrepo_shared.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/createrepo_shared.c b/src/createrepo_shared.c
index 63392d7..5872029 100644
--- a/src/createrepo_shared.c
+++ b/src/createrepo_shared.c
@@ -84,7 +84,7 @@ cr_set_cleanup_handler(const char *lock_dir,
         global_tmp_out_repo = NULL;
 
     // Register on exit cleanup function
-    if (on_exit(failure_exit_cleanup, NULL))
+    if (atexit(failure_exit_cleanup))
         g_warning("Cannot set exit cleanup function by atexit()");
 
     // Prepare signal handler configuration
diff --git a/src/createrepo_shared.h b/src/createrepo_shared.h
index 05abce7..f709a2e 100644
--- a/src/createrepo_shared.h
+++ b/src/createrepo_shared.h
@@ -40,7 +40,7 @@ extern "C" {
 
 /**
  * This function does:
- * 1) Sets on_exit cleanup function that removes temporary directories
+ * 1) Sets atexit cleanup function that removes temporary directories
  * 2) Sets a signal handler for signals that lead to process temination.
  *    (List obtained from the "man 7 signal")
  *    Signals that are ignored (SIGCHILD) or lead just to stop (SIGSTOP, ...)
-- 
2.5.0

